### DISCUSSION TITLE
Future of the test suite

### DISCUSSION THREAD

**Discussion by lgiordani** [grow/grow#1199] [General]
Hi all, I see that Grow uses nose to run tests. Unfortunately nose is currently unmaintained (see https://nose.readthedocs.io/en/latest/) and doesn't support Python 3.10 already. While this is not an immediate issue, it will be in the future.

Are there any plans to move to a different test suite? pytest is a great framework, known to many Python developers, but it would require a considerable effort to move all tests from Unittest. Considerable but far from impossible, with the advantage that it might be done progressively. Other ideas?

Thanks!

Leo

**Comment by jeremydw** [comment]
Hi! We're open to PRs that update the test suite to use a different test runner. Our team won't be working on upgrading Grow as we are migrating projects to use either https://amagaki.dev or Next.js instead, but we'll accept any PRs here. Thanks!

**Comment by lgiordani** [comment]
Hi @jeremydw, thanks for your reply. When you say that the team won't work on upgrading Grow do you mean that it's basically in maintenance mode from your point of view? Are there no more active developers? Thank you

**Reply by jeremydw** [reply]
We are supporting the project in reviewing PRs and fixing bugs, but we (https://github.com/blinkk) aren't adding new features to Grow, since we are moving projects off of Grow and onto Amagaki.dev and/or Next.js.
